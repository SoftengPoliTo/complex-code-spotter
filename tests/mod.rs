use std::path::Path;

use complex_code_spotter::{Complexity, SnippetsProducer};

const SOURCE_PATH: &str = "data/seahorse/src/";
const SNAPSHOT_PATH: &str = "snapshots";

fn run_tests(subdir: &str, complexities: Vec<(Complexity, usize)>) {
    // Snapshot path
    let snapshot_path = Path::new(SNAPSHOT_PATH).join(subdir);

    // Retrieve snippets
    let snippets = SnippetsProducer::new()
        .complexities(complexities)
        .run(Path::new(SOURCE_PATH))
        .unwrap();

    if let Some(snippets) = snippets {
        // Iterate over temporary results
        for snippet in snippets {
            // Snapshot name
            let name = snippet
                .source_path
                .as_path()
                .to_str()
                .unwrap()
                .replace("/", "_");

            insta::with_settings!({
                snapshot_path => &snapshot_path,
                // Sort hashmaps to avoid having different orders in snapshots
                sort_maps => true,
                prepend_module_to_snapshot => false
            },{
                insta::assert_yaml_snapshot!(name, snippet);
            });
        }
    } else {
        assert!(true);
    }
}

#[test]
fn seahorse_high_thresholds() {
    run_tests(
        "high_thresholds",
        vec![(Complexity::Cyclomatic, 15), (Complexity::Cognitive, 15)],
    );
}

#[test]
fn seahorse_low_thresholds() {
    run_tests(
        "low_thresholds",
        vec![(Complexity::Cyclomatic, 1), (Complexity::Cognitive, 1)],
    );
}
