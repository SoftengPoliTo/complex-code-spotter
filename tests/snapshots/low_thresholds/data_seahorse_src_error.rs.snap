---
source: tests/mod.rs
expression: snippet
---
source_path: data/seahorse/src/error.rs
language: Rust
snippets:
  Cyclomatic:
    - complexity: 2
      start_line: 13
      end_line: 23
      text: "impl fmt::Display for FlagError {\n    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {\n        match *self {\n            FlagError::NotFound => f.write_str(\"NotFound\"),\n            FlagError::Undefined => f.write_str(\"Undefined\"),\n            FlagError::TypeError => f.write_str(\"TypeError\"),\n            FlagError::ValueTypeError => f.write_str(\"ValueTypeError\"),\n            FlagError::ArgumentError => f.write_str(\"ArgumentError\"),\n        }\n    }\n}"
    - complexity: 6
      start_line: 14
      end_line: 22
      text: "    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {\n        match *self {\n            FlagError::NotFound => f.write_str(\"NotFound\"),\n            FlagError::Undefined => f.write_str(\"Undefined\"),\n            FlagError::TypeError => f.write_str(\"TypeError\"),\n            FlagError::ValueTypeError => f.write_str(\"ValueTypeError\"),\n            FlagError::ArgumentError => f.write_str(\"ArgumentError\"),\n        }\n    }"
    - complexity: 2
      start_line: 25
      end_line: 35
      text: "impl error::Error for FlagError {\n    fn description(&self) -> &str {\n        match *self {\n            FlagError::NotFound => \"Flag not found\",\n            FlagError::Undefined => \"Flag undefined\",\n            FlagError::TypeError => \"Flag type mismatch\",\n            FlagError::ValueTypeError => \"Value type mismatch\",\n            FlagError::ArgumentError => \"Illegal argument\",\n        }\n    }\n}"
    - complexity: 6
      start_line: 26
      end_line: 34
      text: "    fn description(&self) -> &str {\n        match *self {\n            FlagError::NotFound => \"Flag not found\",\n            FlagError::Undefined => \"Flag undefined\",\n            FlagError::TypeError => \"Flag type mismatch\",\n            FlagError::ValueTypeError => \"Value type mismatch\",\n            FlagError::ArgumentError => \"Illegal argument\",\n        }\n    }"

